dpdk/vr_dpdk_interface.c

```diff
@@ -317,13 +355,14 @@ dpdk_monitoring_if_add(struct vr_interface *vif)
     int ret;
     unsigned short monitored_vif_id = vif->vif_os_idx;
     struct vr_interface *monitored_vif;
+    struct vrouter *router = vrouter_get(vif->vif_rid);
 
     RTE_LOG(INFO, VROUTER, "Adding monitoring vif %u KNI device %s"
                 " to monitor vif %u\n",
                 vif->vif_idx, vif->vif_name, monitored_vif_id);
 
     /* check if vif exist */
-    monitored_vif = vrouter_get_interface(vif->vif_rid, monitored_vif_id);
+    monitored_vif = __vrouter_get_interface(router, monitored_vif_id);
```

Comment:
Don't we need a refcount on monitored vif? What happens if the monitored vif is deleted while it is being monitored?